common/common-client/src/main/java/org/ow2/proactive/authentication/Connection.java

```diff
@@ -162,8 +165,16 @@ public T waitAndConnect(String url, long timeout) throws Exception {
             // waiting until scheduling is initialized
             while (leftTime > 0) {
```

Comment:
I'm not very familiar with all this stuff but don't you think that we can get rid of the whole `while` loop, decrementing `leftTime` variable, `sleepOrThrow` method (thanks to the new `waitFor` call) and just throw `ERROR_NOT_ACTIVATED` if the boolean is false?

Reply:
There are two wait mechanism actually.

1) The PAFuture.waitFor waits for the server to respond to the request (normally it should be fast, but in case there is some problem we add here a timeout).
2) after the server responded we check the status of the server (is it activated or not). If the server is not yet activated, we loop until it is activated (case of server startup).